packages/host/app/resources/search.ts

```diff
@@ -156,53 +159,61 @@ export class Search extends Resource<Args> {
   }
 
   private search = restartableTask(async (query: Query) => {
-    let results = flatMap(
-      await Promise.all(
-        this.realmsToSearch.map(async (realm) => {
-          let json = await this.cardService.fetchJSON(
-            `${realm}_search?${stringify(query)}`,
-          );
-          if (!isCardCollectionDocument(json)) {
-            throw new Error(
-              `The realm search response was not a card collection document:
-        ${JSON.stringify(json, null, 2)}`,
+    // we cannot use the `waitForPromise` test waiter helper as that will cast
+    // the Task instance to a promise which makes it uncancellable. When this is
+    // uncancellable it results in a flaky test.
+    let token = waiter.beginAsync();
```

Comment:
FYI: casting a EC Task to a promise strikes again--please remember to never `await` an EC task outside of the task. Within the task `await` is just syntactic sugar for a `yield`.